---
title: "Seal-micro-R"
author: "Erin D'Agnese"
date: "4/26/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(here)
library(tidyverse)
#devtools::install_github("jbisanz/qiime2R")
library(qiime2R)
```

```{r salmon qiime outputs}
Salmetadata<-read_tsv(here("SalmonSeqs", "salmon.metadata.tsv"))
Salmetadata
SVs<-read_qza(here("SalmonSeqs","table.qza"))
names(SVs)
SVs$uuid
SVs$data[1:5,1:5]
Saltaxonomy<-read_qza(here("SalmonSeqs","taxonomy.qza"))
Saltaxonomy$uuid
Saltaxtable<-Saltaxonomy$data %>% as_tibble() %>% separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species")) #convert the table into a tabular split version
Saltaxtable
Saltree<-read_qza(here("SalmonSeqs","rooted-tree.qza"))
Saltree$uuid
Salshannon<-read_qza(here("SalmonSeqs", "core-metrics-results" ,"shannon_vector.qza"))
Salshannon$uuid
head(Salshannon$data)
Salpco<-read_qza(here("SalmonSeqs", "core-metrics-results" ,"unweighted_unifrac_pcoa_results.qza"))
Salpco$uuid
head(Salpco$data$ProportionExplained)
Salpco$data$Vectors[1:5, 1:3]
SalWpco<-read_qza(here("SalmonSeqs", "core-metrics-results" ,"weighted_unifrac_pcoa_results.qza"))
SalWpco$uuid
head(SalWpco$data$ProportionExplained)
SalWpco$data$Vectors[1:5, 1:3]
Salbc <- read_qza(here("SalmonSeqs", "core-metrics-results", "bray_curtis_distance_matrix.qza"))
SalFaith <-read_qza(here("SalmonSeqs", "core-metrics-results" , "faith_pd_vector.qza"))
SalEven <- read_qza(here("SalmonSeqs", "core-metrics-results" ,"evenness_vector.qza"))
head(SalFaith$data)
head(SalEven$data)
```

```{r seals only qiime outputs}
SOmetadata<-read_tsv(here("Sealseqs","seal.metadata.tsv"))
SOmetadata
SOSVs<-read_qza(here("Sealseqs", "table.qza"))
names(SOSVs)
SOSVs$uuid
SOSVs$data[1:5,1:5]
SOtaxonomy<-read_qza(here("Sealseqs","SILVA138-seal-taxonomy.qza"))
SOtaxonomy$uuid
SOtaxtable<-SOtaxonomy$data %>% as_tibble() %>% separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species")) #convert the table into a tabular split version
SOtaxtable
SOtree<-read_qza(here("Sealseqs","rooted-tree.qza"))
SOtree$uuid
SOshannon<-read_qza(here("Sealseqs", "core-metrics-results","shannon_vector.qza"))
SOshannon$uuid
head(SOshannon$data)
SObc <- read_qza(here("Sealseqs", "core-metrics-results", "bray_curtis_distance_matrix.qza"))
#SOpco<-read_qza("SO-unweighted_unifrac_pcoa_results.qza")
#SOpco$uuid
#head(SSpco$data$ProportionExplained)
#SOpco$data$Vectors[1:5, 1:3]
#SOWpco<-read_qza("SO-weighted_unifrac_pcoa_results.qza")
#SOWpco$uuid
#head(SOWpco$data$ProportionExplained)
#SOWpco$data$Vectors[1:5, 1:3]
#SOFaith <-read_qza("SO-faith_pd_vector.qza")
#SOEven <- read_qza("SO-evenness_vector.qza")
```

```{r make the phyloseq object for seals}
#SOotu<-list.as.matrix()
## THIS IS THE ONLY WAY I WAS ABLE TO CREATE THE PHYLOSEQ TABLE PROPERLY FROM THE QIIME2 - R CREATED ARTIFACTS
library("phyloseq")
OTU = otu_table(SOSVs$data, taxa_are_rows = T)
TAX = tax_table(as.data.frame(SOtaxtable)%>% column_to_rownames("Feature.ID")%>% as.matrix("Taxon"))
TREE = phy_tree(SOtree$data)
SAMPLES = sample_data(SOmetadata %>% as.data.frame() %>% column_to_rownames("sample-id"))
SOphy<- phyloseq(OTU,TAX,SAMPLES)
```

```{r make phyloseq object for salmon}
SalOTU <- otu_table(SVs$data, taxa_are_rows = T)
SalTAX = tax_table(as.data.frame(Saltaxtable)%>% column_to_rownames("Feature.ID")%>% as.matrix("Taxon"))
SalTREE = phy_tree(Saltree$data)
SalSAMPLES = sample_data(Salmetadata %>% as.data.frame() %>% column_to_rownames("sample-id"))
Salphy<- phyloseq(SalOTU,SalTAX,SalSAMPLES)


rank_names(Salphy)
sample_variables(Salphy)
```

Export the otu tables, tax table and metadata into Dataframes to use in downstream analysis outside of phyloseq
```{r}
otu_data = SOasv
otu_id = otu_data$`ASV.id`
otu_data = data.frame(otu_data, check.names = FALSE)
rownames(otu_data) = otu_id

meta_data = SOmetadata
meta_data = meta_data %>% rename(Sample.ID = 'sample-id')

```


Now that both the seal and salmon runs have been processed separately let's merge them into one combined large phyloseq object to use for the salmon + seal metrics - leave TREE out of phyloseq object for this. 
```{r}
CombinedPhy <- merge_phyloseq(SOphy, Salphy)
```

Remove the 


###EVERYTHING BELOW IS THE OLD CODE ONLY PULL FROM IT THEN REMOVE BEFORE PUTTING
ON GITHUB ####
```{r plotting alpha div}
## PLotting alpha div
Salshannon$data %>%
  as.data.frame() %>%
  rownames_to_column("sample-id")%>% # %>%
  left_join(Salmetadata) %>%
  ggplot(aes(x=DemoCat, y=shannon_entropy)) +
  geom_boxplot(outlier.size=2, outlier.shape = 21) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) +
  xlab("group") +
  ylab("Shannon Diversity") 
```

```{r seal shannon div}
SOshannon$data %>%
  as.data.frame() %>%
  rownames_to_column("sample-id")%>% # %>%
  left_join(SOmetadata) %>%
  ggplot(aes(x=DemoCat, y=shannon)) +
  geom_boxplot(outlier.size=2, outlier.shape = 21) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) +
  xlab("Demographic group") +
  ylab("Shannon Diversity") 

SOshannon$data %>%
  as.data.frame() %>%
  rownames_to_column("sample-id")%>% # %>%
  left_join(SOmetadata) %>%
  ggplot(aes(x=Region, y=shannon)) +
  geom_boxplot(outlier.size=2, outlier.shape = 21) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) +
  scale_x_discrete(breaks=c("Bass_strait", "East_coast", "South"), labels=c("Bass Strait", "East Coast", "Southern Region"))+
  xlab("Region") +
  ylab("Shannon Diversity") 
```

```{r salmon faith div}
SalFaith$data %>%
  as.data.frame() %>%
  rownames_to_column("sample-id")%>% # %>%
  left_join(Salmetadata) %>%
  ggplot(aes(x=DemoCat, y=faith_pd)) +
  geom_boxplot(outlier.size=1.5 , outlier.shape = 21) +
  ylim(0, 75) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) +
  xlab("Demographic group") +
  ylab("Faith Phylogentic Diversity") 
```

```{r seal faith div}
SOFaith$data %>%
  as.data.frame() %>%
  rownames_to_column("sample-id")%>% # %>%
  left_join(SOmetadata) %>%
  ggplot(aes(x=DemoCat, y=faith_pd)) +
  geom_boxplot(outlier.size=1.5 , outlier.shape = 21) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) +
  xlab("Demographic group") +
  ylab("Faith Phylogentic Diversity") 

SOFaith$data %>%
  as.data.frame() %>%
  rownames_to_column("sample-id")%>% # %>%
  left_join(SOmetadata) %>%
  ggplot(aes(x=Region, y=faith_pd)) +
  geom_boxplot(outlier.size=1.5 , outlier.shape = 21) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) +
  scale_x_discrete(breaks=c("Bass_strait", "East_coast", "South"), labels=c("Bass Strait", "East Coast", "Southern Region"))+
  xlab("Region") +
  ylab("Faith Phylogentic Diversity") 
```

```{r PCoAs Unifrac}
## Plotting beta div
Salpco$data$Vectors %>%
  rownames_to_column("sample-id") %>% #rename to match the metadata table
  left_join(Salmetadata) %>%
  left_join(Salshannon$data %>% rownames_to_column("sample-id")) %>%
  ggplot(aes(x=PC1, y=PC2, shape=Sample_Species, color=DemoCat, size=shannon)) +
  geom_point() +
  xlab(paste("PC1: ", round(100*Salpco$data$ProportionExplained[1]), "%")) +
  ylab(paste("PC2: ", round(100*Salpco$data$ProportionExplained[2]), "%")) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) 
  ggtitle("Unweighted UniFrac Seal and Salmon at fish farms")


library(gcookbook)
Salpco$data$Vectors %>%
  rename("sample-id"=SampleID) %>% #rename to match the metadata table
  left_join(Salmetadata) %>%
  left_join(Salshannon$data %>% rownames_to_column("sample-id")) %>%
  ggplot(aes(x=PC1, y=PC2, shape=Sample_Species, color=DemoCat, size=shannon)) +
  geom_point() +
  labs(size="Shannon index", color="Demographic cat.", shape = "Species") +
  scale_color_discrete(name = "Demographic cat.", labels=c("Control Adult", 
          "Control Juv.", "In Region Ad.", "Male at FF", "Salmon")) +
  xlab(paste("PC1: ", round(100*Salpco$data$ProportionExplained[1]), "%")) +
  ylab(paste("PC2: ", round(100*Salpco$data$ProportionExplained[2]), "%")) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) 
  ggtitle("Unweighted UniFrac Seal and Salmon at fish farms")

Salpco$data$Vectors %>%
  rename("sample-id"=SampleID) %>% #rename to match the metadata table
  left_join(Salmetadata) %>%
  left_join(Salshannon$data %>% rownames_to_column("sample-id")) %>%
  ggplot(aes(x=PC1, y=PC2, z=PC3, shape=Sample_Species, color=DemoCat, size=shannon)) +
  geom_point() +
  labs(size="Shannon index", color="Demographic cat.", shape = "Species") +
  scale_color_discrete(name = "Demographic cat.", labels=c("Control Adult", 
                                                           "Control Juv.", "In Region Ad.", "Male at FF", "Salmon")) +
  
  xlab(paste("PC1: ", round(100*Salpco$data$ProportionExplained[1]), "%")) +
  ylab(paste("PC2: ", round(100*Salpco$data$ProportionExplained[2]), "%")) +
  theme_bw() 
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) 
  ggtitle("Unweighted UniFrac Seal and Salmon at fish farms")


 SOpco$data$Vectors %>%
  rename("sample-id"=SampleID) %>% #rename to match the metadata table
  left_join(SOmetadata) %>%
  left_join(SOshannon$data %>% rownames_to_column("sample-id")) %>%
  ggplot(aes(x=PC1, y=PC2, shape=AdJuv, color=FishFarm, size=shannon)) +
  geom_point() +
  labs(size="Shannon index", color="At Fish Farm", shape = "Adult or Juv.") +
  scale_color_discrete(name = "At Fish Farm", labels=c("In Region", "No", "Yes")) +
  xlab(paste("PC1: ", round(100*SOpco$data$ProportionExplained[1]), "%")) +
  ylab(paste("PC2: ", round(100*SOpco$data$ProportionExplained[2]), "%")) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) 
  ggtitle("Unweighted UniFrac Juvenile and Adult Seals in relation to Fish Farms")

SOpco$data$Vectors %>%
  rename("sample-id"=SampleID) %>% #rename to match the metadata table
  left_join(SOmetadata) %>%
  left_join(SOshannon$data %>% rownames_to_column("sample-id")) %>%
  ggplot(aes(x=PC1, y=PC2, shape=AdJuv, color=Region, size=shannon)) +
  geom_point() +
  labs(size="Shannon index", color="Region", shape = "Adult or Juv.") +
  scale_color_discrete(name = "Region", labels=c("Bass Strait", "East Coast", "South")) +
  xlab(paste("PC1: ", round(100*SOpco$data$ProportionExplained[1]), "%")) +
  ylab(paste("PC2: ", round(100*SOpco$data$ProportionExplained[2]), "%")) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) +
  ggtitle("Unweighted UniFrac distances between Juvenile and Adult Seals in regions of Tasmania")


SOWpco$data$Vectors %>%
  rename("sample-id"=SampleID) %>% #rename to match the metadata table
  left_join(SOmetadata) %>%
  left_join(SOshannon$data %>% rownames_to_column("sample-id")) %>%
  ggplot(aes(x=PC1, y=PC2, shape=AdJuv, color=FishFarm, size=shannon)) +
  labs(size="Shannon index", color="At Fish Farm", shape = "Adult or Juv.") +
  geom_point() +
  xlab(paste("PC1: ", round(100*SOWpco$data$ProportionExplained[1]), "%")) +
  ylab(paste("PC2: ", round(100*SOWpco$data$ProportionExplained[2]), "%")) +
  theme_bw() +
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) 
  ggtitle("Weighted UniFrac Juvenile and Adult Seals in relation to Fish Farms")



###ordination plots
library(phyloseq)
plot_ordination(SOphy2, ordinate(SOphy2, "DCA"), type ="samples", color="AdJuv")
plot_ordination(SOphy2, ordinate(SOphy2, "DCA"), type ="samples", color="FishFarm")



###network
SOnet<-make_network(SOphy, max.dist = 0.3)
SOnet<-make_network(SOphy, max.dist = "bray")
plot_network(SOnet, SOphy, color="AdJuv", shape = "FishFarm", label = NULL)
plot_network(SOnet, SOphy, type = "taxa", color="AdJuv", shape = "FishFarm", label = NULL)

plot_network(SOnet, SOphy, type = "taxa", color= "Class", 
    point_size=4, alpha=1, label="Genus", hjust=1.35, 
    line_weight=0.5, line_color=color, line_alpha=0.4,
    title=NULL)


###Phyloseq tutorial
sample_names(SOphy)
rank_names(SOphy)
sample_variables(SOphy)
##subsets of data for analysis within groups
Juvphy <-subset_samples(SOphy, AdJuv=="Juv")
Adphy <-subset_samples(SOphy, AdJuv=="Ad")
FFphy <-subset_samples(SOphy, FishFarm=="yes")
Contphy <-subset_samples(SOphy, FishFarm=="no")
Soutphy <-subset_samples(SOphy, Region=="South")
BSphy <-subset_samples(SOphy, Region=="Bass_strait")
BactSOphy <-subset_taxa(SOphy, Kingdom=="Bacteria")
PrteoSOphy <- subset_taxa(SOphy, Phylum=="Proteobacteria")
FusSOphy <- subset_taxa(SOphy, Phylum=="Fusobacteria")
FirmSOphy <-subset_taxa(SOphy, Phylum=="Firmicutes")

BactSalphy <-subset_taxa(Salphy, Kingdom=="Bacteria")

### removing NAs
data(SOphy)
gpSO <- tax_glom(SOphy, taxrank="Phylum")  # 7 taxonomic ranks
head(tax_table(gpSO))   # last columns are filled with NAs

# Remove taxonomic levels filled with NAs
gp2SO <- phyloseq_rm_na_tax(SOphy)  # 5 taxonomic ranks
head(tax_table(gp2SO))
head(tax_table(SOphy))

#### full data phyloseq code 
## normalize number of reads in each sample using median seq depth
total = median(sample_sums(SOphy))
standf = function(x, t=total) round(t *(x/sum(x)))
NmSOphy = transform_sample_counts(SOphy, standf)

Saltotal = median(sample_sums(Salphy))
Salstandf = function(x, t=Saltotal) round(t *(x/sum(x)))
NmSalphy = transform_sample_counts(Salphy, Salstandf)
### data was already normalized it looks like as the values appear the same
plot_bar(SOphy, fill="Class")
SOphy
NmSOphy
plot_bar(NmSOphy)
plot_bar(BactSOphy, x="FishFarm")

plot_bar(Salphy, fill="Class")
SSphy
NmSSphy
plot_bar(NmSSphy, x="DemoCat")

plot_bar(BactSOphy, x="FishFarm")

##heatmap
plot_heatmap(SOphy, method = "NMDS", distance = "bray")
SOphy_abund <- filter_taxa(SOphy, function(x) sum(x > total*0.10)>0, TRUE)
SOphy_abund
BactSOphy_abund<-filter_taxa(BactSOphy,function(x) sum(x > total*0.20)>0, TRUE)
plot_heatmap(SOphy_abund, method = "NMDS", distance = "bray")
SOphy_20abund <- filter_taxa(SOphy2, function(x) sum(x > total*0.20)>0, TRUE)
plot_heatmap(SOphy_20abund, method = "NMDS", distance = "bray", sample.label = "AltIdentifier", taxa.label = "Phylum", taxa.order = "Phylum")

plot_heatmap(SOphy_20abund, method = "NMDS", distance = "bray", sample.label = "AltIdentifier", taxa.label = "Class")

plot_net(SOphy, distance = "bray", maxdist = 0.5, type = "taxa", color="Phylum", point_label = "Class")

plot_net(SOphy, distance = "bray", type = "taxa", laymeth = "kamada.kawai" , maxdist = 0.7, color="Class")

plot_net(SOphy, distance = "bray", type = "taxa", laymeth = "svd" , maxdist = 0.7, color="Class")
plot_net(SOphy, distance = "bray", type = "taxa", laymeth = "fruchterman.reingold" , maxdist = 0.7, color="Class")
plot_net(SOphy, distance = "bray", type = "taxa", laymeth = "graphopt" , maxdist = 0.7, color="Class")


##this network worked, so the class and phylym stuff is all correctly represented in the taxtable now. 

plot_net(SOphy, distance = "(A+B-2*J)/(A+B)", type="taxa", maxdist=0.8, color="Class", point_label = "Family")

distanceMethodList

plot_net(SSphy, distance = "bray", maxdist=0.7, color="SmFFAdJuvCAd") +
  labs(color="Demographic") +
  scale_color_discrete(name = "Demographic", labels=c("Control Adults", "Control Juv.", "In Region Adults", "Males at FF", "Salmon"))
##output figure = SS-network-demo-bray-0.7.png
plot_net(SSphy, distance = "bray", maxdist=0.6, color="SmFFAdJuvCAd") +
  labs(color="Demographic") +
  scale_color_discrete(name = "Demographic", labels=c("Control Adults", "Control Juv.", "In Region Adults", "Males at FF", "Salmon"))
##output figure = SS-network-demo-bray-0.6.png
plot_net(SSphy, distance = "unifrac", maxdist=0.8, color="SmFFAdJuvCAd") +
  labs(color="Demographic") +
  scale_color_discrete(name = "Demographic", labels=c("Control Adults", "Control Juv.", "In Region Adults", "Males at FF", "Salmon"))
##output figure = SS-network-demo-sp-unifrac-0.8.png
plot_net(SOphy, distance = "bray", maxdist=0.6, color="DemoCat")+ 
  labs(color="Group of seals")+ 
  scale_color_discrete(name = "Group of seals", labels=c("Control Ad.", "Control Juv.", "In Region Ad.", "Males at FF"))
 


### Ordination
plot_bar(SOphy, fill = "Class") + 
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack")

##regroup juvs and adults
SOphy_AdJuv <- merge_samples(SOphy, "AdJuv")
plot_bar(SOphy_AdJuv, fill = "Class")+
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack")


SOphy_FF <- merge_samples(SOphy, "FishFarm")
plot_bar(SOphy_FF, fill = "Class")+
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack")

plot_bar(SOphy, x="DemoCat", fill="Class") +
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack")+
  theme_bw()+
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank()) 
  



plot_bar(SSphy, x="Class", fill="Class", facet_grid = Sample_Species~DemoCat) +
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack")

SSphy2_Sp <- merge_samples(SSphy, "Sample_Species")
SSphy_Dm <-mege_samples(SSphy, "DemoCat")
plot_bar(SSphy2_Sp, fill = "Class")+
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack")

### relative abundancy plots
ps<- tax_glom(SSphy, taxrank=rank_names(SSphy), NArm=TRUE, bad_empty=c(NA,"", " ", "\t"))
ps0<- transform_sample_counts(ps, function(x) x/ sum(x))
ps1<- merge_samples(ps0, "DemoCat")
ps2<- transform_sample_counts(ps1, function(x) x/ sum(x))
plot_bar(ps2, fill="Phylum")
### everytime I transform the salmon abundance isn't calculated or shown in the fig
nps<- tax_glom(NSSphy, "Phylum")
nps0<- transform_sample_counts(nps, function(x) x/ sum(x))
nps1<- merge_samples(NmSSphy, "Sample_Species")
nps2<- transform_sample_counts(nps1, function(x) x/ sum(x))
plot_bar(nps2, fill="Phylum")

Pps<- tax_glom(SSphy, "Phylum")
Pps0<- transform_sample_counts(Pps, function(x) x/ sum(x))
Pps1<- merge_samples(SSphy, "Sample_Species")
Pps2<- transform_sample_counts(Pps1, function(x) x/ sum(x))
plot_bar(Pps2, fill="Phylum")

plot_bar(Pps2, fill = "Phylum")+
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack")+
  theme_bw()+theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())

BPps<- tax_glom(BactSSphy, "Phylum")
BPps0<- transform_sample_counts(BPps, function(x) x/ sum(x))
BPps1<- merge_samples(BactSSphy, "Sample_Species")
BPps2<- transform_sample_counts(BPps1, function(x) x/ sum(x))
plot_bar(BPps2, fill="Phylum")

BPpso<- tax_glom(BactSOphy, "Phylum")
BPpso0<- transform_sample_counts(BPpso, function(x) x/ sum(x))
BPpso1<- merge_samples(BactSOphy, "AdJuv")
BPpso2<- transform_sample_counts(BPpso1, function(x) x/ sum(x))
BPDso1<- merge_samples(BactSOphy, "DemoCat")
BPDso2<- transform_sample_counts(BPDso1, function(x) x/ sum(x))

PbSO <- tax_glom(PrteoSOphy, "Family")
PbSO1 <- merge_samples(PrteoSOphy, "DemoCat")
PbSO2 <- transform_sample_counts(PbSO1, function(x) x/ sum(x))

FbSO <- tax_glom(FirmSOphy, "Family")
FbSO1 <- merge_samples(FirmSOphy, "DemoCat")
FbSO2 <- transform_sample_counts(FbSO1, function(x) x/ sum(x))

plot_bar(BPps2, fill = "Phylum")+
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack")+
  theme_bw()+ theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())

plot_bar(BPpso2, fill = "Phylum")+
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack")+
  theme_bw()+ theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())


BPps<- tax_glom(BactSSphy, "Phylum")
BPps0<- transform_sample_counts(BPps, function(x) x/ sum(x))
BPps1<- merge_samples(BactSSphy, "Sample_Species")
BPps2<- transform_sample_counts(BPps1, function(x) x/ sum(x))


Bdps1 <-merge_samples(BactSSphy, "DemoCat")
Bdps2<- transform_sample_counts(Bdps1, function(x) x/ sum(x))


plot_bar(Bdps2, fill = "Phylum")+
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack")+
  theme_bw()+ theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())
  
  cale_fill_brewer(colorRampPalette(brewer.pal(12, "Accent"))(colourCount))                    

plot_bar(BPDso2, fill = "Phylum")+
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack")+
  theme_bw()+ 
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())

plot_bar(PbSO2, fill = "Family")+
  geom_bar(aes(color=Family, fill=Family), stat="identity", position="stack")+
  theme_bw()+ 
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())

plot_bar(FbSO2, fill = "Family")+
  geom_bar(aes(color=Family, fill=Family), stat="identity", position="stack")+
  theme_bw()+ 
  theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())
  
  scale_color_discrete(names="Phylum", labels=c("Actinobacteria", "Bacteriodetes", "Chloroflexi", "Cyanobacteria",
      "Deferribacteres", "Dependentiae", "Epsilonbacteraeota", "Firmicutes",
      "Fusobacteria", "Marinimicrobia", "Patescibacteria", "Planctomycetes",
      "Proteobacteria", "spirochaetes", "Tenericutes", "Verrucomicrobia", "NA"))

plot_bar(BPpso2, fill = "Phylum")+
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack")+
  theme_bw()+ theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())

plot_bar(BPpso2, fill = "Class")+
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack")+
  theme_bw()+ theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())


plot_bar(Bdps2, fill = "Class")+
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack")+
  theme_bw()+ theme(panel.grid.major=element_blank(), panel.grid.minor=element_blank())


NmSSphy
SSphy2fit <- filter_taxa(SSphy, function(x=Sample) mean(x=Sample) >0.5, TRUE)

SSphyRel<-transform_sample_counts(SSphy2fit, function(x)x/sum(x))
SSphyRel

SSphyRel_Sp <- merge_samples(SSphyRel, "Species")
plot_bar(SSphyRel, fill = "Class")+
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack")

SSphyRel_Ph <- merge_samples(SSphyRel, "Species")
plot_bar(SSphyRel_Ph, fill = "Phylum")+
  geom_bar(aes(color=Phylum, fill=Phylum), stat="identity", position="stack")

plot_bar(SSphyRel, x="Class", fill="Class", facet_grid = Species~DemoCat) +
  geom_bar(aes(color=Class, fill=Class), stat="identity", position="stack")

###Alpha div
plot_richness(SOPhy2, measures=c("Chao1", "Shannon"), x="AdJuv", color="FishFarm")
plot_richness(SOPhy2, measures=c("Chao1", "Shannon"), x="FishFarm", color="AdJuv")

### vegan 
install.packages("vegan")
library(vegan)
devtools::install_github("gavinsimpson/ggvegan")
##piping phyloseq objects into vegan


### try this altenate - this worked
otu <- otu_table(SSphy)
if (taxa_are_rows(otu)) {
  otu <- t(otu)
}
SSmat <- as(otu, "matrix")
SSdf <- as.data.frame(SSmat)

SSsampsdf <- as(sample_data(SSphy), "data.frame")

# convert the sample_data() within a phyloseq object to a vegan compatible data object
pssd2veg <- function(SSphy) {
  sd <- sample_data(SSphy)
  return(as(sd,"data.frame"))
}
pssd2veg <- function(SSphy) 
  sd <- sample_data(SSphy)
return(as(sd,"data.frame"))
# convert the otu_table() within a phyloseq object to a vegan compatible data object
psotu2veg <- function(SSphy) {
  OTU <- otu_table(SSphy)
  if (taxa_are_rows(OTU)) {
    OTU <- t(OTU)
  }
  return(as(OTU, "matrix"))
}

# move an OTU table from vegan to phyloseq  
otu_table(PhyloseqObject) <- otu_table(veganOTUobject, taxa_are_rows=TRUE)  
# move sample data from vegan to phyloseq
sample_data(PhyloseqObject) <- as.data.frame(veganSampleDataObject)

### Heirarchical clustering in vegan
SSdistbray <- vegdist(SSdf, "bray")
SScsin <- hclust(SSdistbray, method="single")

install.packages("compositions")
###centered log ratio tansform of OTU table to deal with the zero-inflated OTU table
library(compositions)
SSmatCLR <- clr(SSmat)

library(vegan)
SScsin <- hclust(SSmatCLR, method="single")

SSeuc <- vegdist(SSdf, "euclidean")
chord_SS <- vegdist(decostand(SSdf, "norm"),"euclidean")

SS.single.euc<-hclust(SSeuc, method="single")
SS.single.chord<-hclust(chord_SS, method="single")
plot(SS.single.euc, hang=1)
rect.hclust(SS.single.euc, 3)
plot(SS.single.chord)
rect.hclust(SS.single.chord,3)

SScom<-hclust(chord_SS, method="complete")
library(clustsig)
simprof(SSdf, num.expected = 1000, num.simulated = 999,
        method.cluster="average", method.distance="euclidean",
        method.transform = "identity", alpha=0.05, 
        sample.orientation = "row", const=0, 
        silent=TRUE, increment = 100, undef.zero = TRUE, warn.braycurtis = TRUE)
simprof.plot(results, leafcolors=NA, plot=TRUE, fill=TRUE,
             leaflab="perpendicular", siglinetype=1)

plot(SScom, hang=1)
rect.hclust(SScom, 3)
SSaver<-hclust(chord_SS, method="aver")
plot(SSaver, hang=1)
rect.hclust(SSaver, 3)

install.packages("ape")
library(ape)
plot(as.phylo(SSaver), type="unrooted", cex=0.6, 
     no.margin=TRUE, label.offset = 0.1, tip.color=colors[clus5])
colors=c("red","blue","green","black","purple")
clus5<-cutree(SSaver, 5)

plot(as.phylo(SScom), type="fan", collapse=TRUE, no.margin=TRUE, cex= 0.4, label.offset = 0.05, tip.color=colors[clus5com])

plot(as.phylo(SScom), type="fan", cex=0.6, 
     no.margin=TRUE,, label.offset = 0.1, tip.color=colors[clus5com]) + 
  tiplabels(AltIdentifier)
colors=c("red","blue","green","black","purple")
nodelabels("AltIdentifier")
clus5com<-cutree(SScom, 5)

plot(as.phylo(SSaver), type="fan", cex=0.6, 
     no.margin=TRUE, label.offset = 0.1)

### need to sort out what to do downstream of vegemite
SSvegclass<- vegemite(SSdf, SSaver, scale="log")

install.packages("dendextend")
install.packages("circlize")

library(dendextend)
library(circlize)

hc <- hclust(dist(SSaver))
dend <- as.dendrogram(hc)

num_clades <- 5

dend <- dend %>% 
  color_branches(k=num_clades, col=rainbow) %>% 
  color_labels(k=num_clades, col=rainbow)

par(mar = rep(0, 4))
circlize_dendrogram(dend, dend_track_height = 0.8) 


##indicator analysis
library(indicspecies)
groups = c(rep(1:299), rep(301:356), rep(357:409), rep(410:470), rep(471:521))
SOgroups = c(rep(1:56), rep(57:109), rep(110:170), rep(171:221))

TssISA <- read.csv(file="SS-ISA-OTU-Table.csv", sep=",", header=TRUE)
TsoISA <- read.csv(file="SO-OTU-ISA-table.csv", sep=",", header=TRUE)
ssISA<- as.data.frame(TssISA, row.names = NULL, optional=FALSE)
data(ssISA)
TssISA

indval = multipatt(TssISA, groups, control=how(nperm=999))
summary(indval)

SOindval = multipatt(TsoISA, SOgroups, control=how(nperm=999))
summary(SOindval)


##need to get OTU table at matrix with rows as samples and columns as taxa
OTU1 = as(otu_table(SSphy), "matrix")
if(taxa_are_rows(SSphy)){OTU1<-t(OTU1)}
OTUdf = as.data.frame(OTU1)
groups = kmeans(OTU1, centers = 3)
head(OTUdf)





SOOTU1 = as(otu_table(SOhy), "matrix")
if(taxa_are_rows(SOphy)){SOOTU1<-t(SOOTU1)}
OTUdf = as.data.frame(SOOTU1)
groups = 
head(OTUdf)
indval = multipatt(OTUdf, groups,func = "IndVal.g", min.order = 1,control=how(nperms=999))



###SIMPROF
library(clustsig)
help(clustsig)
??clustsig

simprof(, num.expected=1000, num.simulated=999,
        method.cluster="average",
        method.distance="euclidean", 
        method.transform="identity", alpha=0.05,
        sample.orientation="row", const=0,
        silent=TRUE, increment=100,
        undef.zero=TRUE, warn.braycurtis=TRUE)
```

